libs/analytics/migrations/202104281912.do.0078-contactType.sql

```diff
@@ -0,0 +1,56 @@
+-- Recreates the view only to add the new column contact_type
+DROP MATERIALIZED VIEW IF EXISTS contactview_metadata CASCADE;
+CREATE MATERIALIZED VIEW contactview_metadata AS
+  SELECT doc->>'_id' AS uuid,
+    doc->>'name' AS name,
+    doc->>'type' AS type,
+    doc->>'contact_type' AS contact_type,       --> only this is new
+    doc#>>'{contact,_id}' AS contact_uuid,
+    doc#>>'{parent,_id}' AS parent_uuid,
+    doc->>'notes' AS notes,
+    TIMESTAMP WITH TIME ZONE 'epoch' + (doc->>'reported_date')::numeric / 1000 * interval '1 second' AS reported
+  FROM raw_contacts;
+
+CREATE UNIQUE INDEX contactview_metadata_uuid ON contactview_metadata (uuid);
+CREATE INDEX contactview_metadata_contact_uuid ON contactview_metadata (contact_uuid);
+CREATE INDEX contactview_metadata_parent_uuid ON contactview_metadata (parent_uuid);
+CREATE INDEX contactview_metadata_type ON contactview_metadata (type);
+
+-- NOTE: The recreation of the view above caused 4 other views to be dropped in cascade,
+--       here are the scripts to recreate them:
+
+CREATE VIEW contactview_hospital AS
```

Comment:
Yes, next time we use Postgrator to mantain views, may be better to use *.js script to reuse better the SQL scripts.

Also it is sad that Postgres does not allow to add a column in a view without the need to recreate it manually, whether the engine needs to recreate the entirely view or not, it should be an internal decision made by the engine.

Reply:
I actually read a little around this, and people were complaining that they can't update materialized views in pgadmin - which is arguably probably even more popular than Postgrator. It could be pgadmin has improved since those posts though, but my point is that this seems like it is a common problem with other pg tools and not something that's easily automated.

Reply:
Yes the problem is that Postgres does not support `ALTER VIEW xxx ADD COLUMN ...` like in tables does. It does allow to do some `ALTER` changes but the most basic one that is to add a column is not supported.

Reply:
... and my point is that even if that is not possible without the recreation of the whole materialized view, that should be an implementation detail for the engine, not for the user, Postgres could support `ALTER VIEW xxx ADD COLUMN ...` syntax, adding a note in the documentation, like: _if you add a column in a materialized view internally the engine will recreate the whole view and dependent views and indexes, and that can take a while, use with precaution ..._

Reply:
yea, I agree it's a less then ideal situation.